### ISSUE TITLE
Fix S1172 FP: Don't report on partial method parameters

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#8988]
Originally reported at https://community.sonarsource.com/t/false-positive-for-s1172/112108

### Description

> If a partial method is defined within generated code that is emitted to the “obj” folder, a false positive may occur for rule S1172. When implementing a partial method it may not be necessary to use all parameters, but it is necessary to define them all so that partial method declarations match.

### Repro steps

```cs
public partial class Test
{
    private partial string FalsePositive(string one, string two);
}

public partial class Test
{
    // S1172 Remove this unused method parameter 'one'. (https://rules.sonarsource.com/csharp/RSPEC-1172)
    private partial string FalsePositive(string one, string two) => two;
}
```
### Expected behavior

Partial methods should be excluded from reporting. This applies to both "old" partial methods and [extended partial methods](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/extending-partial-methods).

### Actual behavior

FP: An issue is raised on partial method implementations.

### Known workarounds

None

**Comment by pavel-mikula-sonarsource** [comment]
https://sonarsource.atlassian.net/browse/NET-863
